pkg/config/tlscfg/flags.go

```diff
@@ -70,31 +70,35 @@ func (c ServerFlagsConfig) AddFlags(flags *flag.FlagSet) {
 }
 
 // InitFromViper creates tls.Config populated with values retrieved from Viper.
-func (c ClientFlagsConfig) InitFromViper(v *viper.Viper) Options {
+func (c ClientFlagsConfig) InitFromViper(v *viper.Viper) (Options, error) {
 	var p Options
-	if c.ShowEnabled {
-		p.Enabled = v.GetBool(c.Prefix + tlsEnabled)
-	}
 	p.CAPath = v.GetString(c.Prefix + tlsCA)
 	p.CertPath = v.GetString(c.Prefix + tlsCert)
 	p.KeyPath = v.GetString(c.Prefix + tlsKey)
 	if c.ShowServerName {
 		p.ServerName = v.GetString(c.Prefix + tlsServerName)
 	}
 	p.SkipHostVerify = v.GetBool(c.Prefix + tlsSkipHostVerify)
-	return p
+	if c.ShowEnabled {
+		p.Enabled = v.GetBool(c.Prefix + tlsEnabled)
+	} else {
+		return p, fmt.Errorf("%s has been set to False", c.Prefix+tlsEnabled)
```

Comment:
* `c.ShowEnabled == false` doesn't mean that tls was disabled, it means that the CLI option does not need to be present
* even if TLS is disabled (`p.Enable == false`), doesn't mean that this is an error.  The error is when it is false but any other flags are non-empty: `(!p.Enabled || !c.ShowEnabled) && anyOtherFieldIsNotEmpty`